internal/auth/authn_repo_test.go

```diff
@@ -19,20 +19,51 @@ func TestRepositoryAuthenticator(t *testing.T) {
 
 	require.NoError(t, repo.WriteSession(ctx, env.Repository, repo.WriteSessionOptions{},
 		func(ctx context.Context, w repo.RepositoryWriter) error {
-			p := &user.Profile{
-				Username: "user1@host1",
+			for _, tc := range []struct {
+				profile  *user.Profile
+				password string
+			}{
+				{
+					profile: &user.Profile{
+						Username:            "user1@host1",
+						PasswordHashVersion: crypto.HashVersion1,
+					},
+					password: "password1",
+				},
+				{
+					profile: &user.Profile{
+						Username:               "user2@host2",
+						KeyDerivationAlgorithm: crypto.ScryptAlgorithm,
+					},
+					password: "password2",
+				},
+				{
+					profile: &user.Profile{
+						Username: "user3@host3",
+					},
+					password: "password3",
+				},
```

Comment:
Lets also add the new algorithm in the test cases here:

```suggestion
				},
				{
					profile: &user.Profile{
						Username:               "user4@host4",
						KeyDerivationAlgorithm: crypto.Pbkdf2Algorithm,
					},
					password: "password4",
				},
```